azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py

```diff
@@ -199,3 +200,37 @@ def _setup_instrumentations(configurations: Dict[str, ConfigurationValue]):
                 lib_name,
                 exc_info=ex,
             )
+
+
+def _is_instance_or_none(var, type):
+    return isinstance(var, type) or var is None
+
+
+def _validate_configurations(configurations):
+    assert _is_instance_or_none(configurations.get("connection_string"), str)
+    assert _is_instance_or_none(
+        configurations.get("exclude_instrumentations"), Sequence
+    )
+    assert _is_instance_or_none(configurations.get("resource"), Resource)
```

Comment:
The idea of this PR is not really to do type checking (it would be too cumbersome to always check for typing, plus this is why we use typing.

We want to only have a few validation checks for logic (like interval needs to be > 0).

The idea for Python and for the distro is that if input is not the correct type, it will not do anything with that input.